replay_processing/sc2files.py

```diff
@@ -95,10 +94,11 @@ def extract_player_data(result, player, logger, debug=False):
     return data
 
 
-def dump(data):
+def dump(data, outfile):
```

Comment:
Another concern I have is that we are writing data within our threads. I was planning on moving this function call out of the thread worker, and instead just use the Future the worker returns to get the data into a list. We can then iterate the list and write the file once, without holding up our workers.